dali/test/python/reader/test_fits.py

```diff
@@ -12,21 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import itertools
+import os
+import sys
+
 from nvidia.dali import pipeline_def
 import nvidia.dali.fn as fn
-import os
 from astropy.io import fits
 import numpy as np
 import tempfile
 import random
+from nose2.tools import params
 from test_utils import to_array
 from numpy.testing import assert_array_equal
 
-rng = np.random.RandomState(12345)
 
+def enum_product(*params):
```

Comment:
Please don't. Use `cartesian_params`.

Reply:
But then I don't have the enumeration.

Reply:
Good point. Do we need enumeration after all?

Reply:
The tests were written so that they randomly select some paramters (`random.choice` and np's random generator declared globally used for generating ndarray). Depending on the order or subset of tests called, the random choices would be different. To avoid that, I create the random generators locally in each test case. The enumeration is used as a seed.